langchain/vectorstores/faiss.py

```diff
@@ -202,19 +203,22 @@ def from_texts(
         return cls(embedding.embed_query, index, docstore, index_to_id)
 
     def save_local(self, path: str) -> None:
-        """Save FAISS index to disk.
+        """Save FAISS index, docstore, and index_to_docstore_id to disk.
 
         Args:
-            path: Path to save FAISS index to.
+            path: .pkl path to save index, docstore, and index_to_docstore_id to.
         """
-        faiss = dependable_faiss_import()
-        faiss.write_index(self.index, path)
+        print(type(self.index), type(self.docstore), type(self.index_to_docstore_id))
+        pickle.dump(
```

Comment:
It looks to me like this may write the index memory pointer, i.e. the content of the variable, rather than the index content. This would manifest as a failure reading back the correct data from a newly launched process. There are index writing methods at https://github.com/facebookresearch/faiss/wiki/Index-IO,-cloning-and-hyper-parameter-tuning#generic-io-support .

Reply:
Just pushed some more code to create a path that stores index and the other variables separately. Kind of a hack-y fix but it works for now.